.github/workflows/build_and_test.yml

```diff
@@ -1293,7 +1293,7 @@ jobs:
           done
 
   pin-build-tools-image:
-    needs: [ build-build-tools-image, push-compute-image-prod, push-neon-image-prod, build-and-test-locally ]
+    needs: [ build-build-tools-image, test-images, build-and-test-locally ]
```

Comment:
These dependencies didn't make much sense to me tbqh. I've tried following the history of this job and traced it back to it's origins in https://github.com/neondatabase/neon/pull/6795, where it already depended on `promote-images`. @bayandin, is it fine to make this depend on test-images only?

Reply:
The initial idea was to call `pin-build-tools-image` once we sure the new version works. Having it after `test-images` totally makes sense